### DISCUSSION TITLE
Why are page and image dimensions different after conversion?

### DISCUSSION THREAD

**Discussion by jscodecode** [pymupdf/PyMuPDF#2215] [Q&A]
` 
    doc = fitz.open()

   #1. open image as fitz.Document and convert it to a PDF in memory

   imgpdf = fitz.open('1.png').convert_to_pdf()

   #2. append memory PDF as a page

   doc.insert_pdf(fitz.open("pdf", imgpdf))

   doc.save('1.pdf')
`

image size is changed from w:{595 },h:{842}(1.png size) to w:{447},h:{632}(1.pdf size)

How can I solve this problems?

**Comment by JorjMcKie** [comment]
This is a typical "Discussions" post not an issue, therefore I will first transfer it.

**Comment by JorjMcKie** [comment]
Everything works as it should.
Look at the image's DPI, you will find an explanation there.
The page rectangle in the document is computed as `fitz.Rect(0, 0, width * 72 / DPI, height * 72 / DPI)`, because its dimension is measured in points, where 1 inch = 72 points.
Here is an example from an image with 96 DPI. Please follow the argument by looking at the numbers in this picture:
![grafik](https://user-images.githubusercontent.com/8290722/216959463-160d93b3-6a57-4f0d-ad1e-6aaab20dc8ac.png)
Multiply the "print size" inch numbers with 72 and you will see which page dimension is computed by PyMuPDF - **_it is not 439 x 501!_**

If you absolutely want that, you need different code:
```python
pix = fitz.Pixmap(imagefile)
page = doc.new_page(width=pix.width, height=pix.height)
page.insert_image(page.rect, pixmap=pix)
```

But please remember: the DPI are there for image quality reasons. If you ignore this as shown, the image on that page will naturally appear with a lower quality.

**Reply by jscodecode** [reply]
I have solved this problem with a higher quality by your answer.  I sincerely appreciate your help.
